### DISCUSSION TITLE
Why can't I change the center of gravity or center of buoyancy for a hydrodynamic body in WEC-Sim?

### DISCUSSION THREAD

**Discussion by jtgrasb** [WEC-Sim/WEC-Sim#1354] [Q&A] [answered]
## Why can't users change the center of gravity (CG) or center of buoyancy (CB) for a hydrodynamic body in WEC-Sim?

When you try to change the CG or CB for a hydrodynamic body in WEC-Sim, you will get the warning: 
_Center of gravity and center of buoyancy are overwritten by h5 data for hydro bodies._ 
The CG and CB from the h5 files are always maintained even if you try to specify something different in wecSimInputFile.m. The reason we do not allow the CG or CB to be changed from the input file is because a change in the CG or CB implies a change in the hydrodynamic coefficients. 
- WEC-Sim requires that all hydrodynamic coefficients are calculated in BEM with respect to the CG. Any change in CG would impact these calculations.
- Although a change in CB alone may not impact the hydrodynamic coefficients, the only way the CB changes is through a change in the displaced volume. If the displaced volume changes, the hydrodynamic coefficients will also change.
Therefore, any change in CG or CB should be accompanied by a new BEM run.

## What cases would modifying the CG or CB be okay without rerunning BEM?
- Examples?

## How can the CG and CB be modified without rerunning BEM?
If you are confident that you can modify your CG or CB without impacting the hydrodynamic coefficients, the best practice is to modify the h5 file in which these parameters are stored. This can be done by either modifying the file directly or rerunning bemio.m and manually setting `hydro.cg` or `hydro.cb` before saving as an h5 file.

**Comment by kmruehl** [comment, marked as answer]
See above responses
